components/suggestion/switch_channel_provider.jsx

```diff
@@ -325,4 +346,40 @@ export default class SwitchChannelProvider extends Provider {
             });
         }, 0);
     }
+
+    formatUnreadChannelsAndDispatch(suggestionId) {
+        const getChannel = makeGetChannel();
+
+        const unreadChannelIds = getSortedUnreadChannelIds(getState(), false);
+
+        const channels = [];
+        for (let i = 0; i < unreadChannelIds.length; i++) {
+            const channel = getChannel(getState(), {id: unreadChannelIds[i]}) || {};
+
+            let wrappedChannel = {channel, name: channel.name, deactivated: false};
+            if (channel.type === Constants.GM_CHANNEL) {
+                wrappedChannel.name = getChannelDisplayName(channel);
+            } else if (channel.type === Constants.DM_CHANNEL) {
+                wrappedChannel = this.userWrappedChannel(
+                    UserStore.getProfile(Utils.getUserIdFromChannelId(channel.name))
```

Comment:
We are in the process of migrating components into redux-style. To make it easier, we are avoiding the use of `*Store` such as `UserStore`. Instead of this, I recommend to use selector `getUser` from `import {getUser} from 'mattermost-redux/selectors/entities/users';`